### PR TITLE
feat: Add datafile accessor

### PR DESCRIPTION
**Pull request by mnoman09**
Summary
-------
- Add datafile accessor methods/functionalities

Testing
-------
- All unit tests and FSC must pass
https://travis-ci.com/github/optimizely/fullstack-sdk-compatibility-suite/builds/181371500

### CODE REVIEW

**Comment by msohailhussain** [comment]
unit test missing. toDatafile

File: core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfig.java [hunk 1]
```diff
@@ -46,14 +59,20 @@ public String getRevision() {
...
     @Override
     public boolean equals(Object obj) {
         if (obj == null || getClass() != obj.getClass()) return false;
         if (obj == this) return true;
         OptimizelyConfig optimizelyConfig = (OptimizelyConfig) obj;
         return revision.equals(optimizelyConfig.getRevision()) &&
             experimentsMap.equals(optimizelyConfig.getExperimentsMap()) &&
-            featuresMap.equals(optimizelyConfig.getFeaturesMap());
+            featuresMap.equals(optimizelyConfig.getFeaturesMap()) &&
+            Objects.equals(datafile, optimizelyConfig.getDatafile());
```
Comment (jaeopt) [inline core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfig.java:75]: Do we need include "datafile" string compare?  String with formatting spaces, etc. will make compare fal. 
It should be enough to compare the rest of fields for OptimizelyConfig equality.

File: core-api/src/main/java/com/optimizely/ab/config/DatafileProjectConfig.java [hunk 2]
```diff
@@ -480,7 +487,8 @@ public ProjectConfig build() throws ConfigParseException {
                 throw new ConfigParseException("Unable to parse empty datafile.");
             }
 
-            ProjectConfig projectConfig = DefaultConfigParser.getInstance().parseProjectConfig(datafile);
+            DatafileProjectConfig projectConfig = (DatafileProjectConfig) DefaultConfigParser.getInstance().parseProjectConfig(datafile);
```
Comment (jaeopt) [inline core-api/src/main/java/com/optimizely/ab/config/DatafileProjectConfig.java:490]: For safety, can we use "ProjectConfig" type and check instanceof before assigning datafile.
```suggestion
ProjectConfig projectConfig = DefaultConfigParser.getInstance().parseProjectConfig(datafile);
if (projectConfig instanceof DatafileProjectConfig) {
   ((DatafileProjectConfig)projectConfig).datafile = datafile;
```

File: core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfig.java [hunk 3]
```diff
@@ -46,14 +59,20 @@ public String getRevision() {
         return revision;
     }
 
+    @JsonIgnore
```
Comment (jaeopt) [inline core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfig.java:62]: Is this @JsonIgnore redundant here?

File: core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfig.java [hunk 4]
```diff
@@ -27,8 +26,6 @@
     private Map<String, OptimizelyExperiment> experimentsMap;
     private Map<String, OptimizelyFeature> featuresMap;
     private String revision;
-
-    @JsonIgnore
```
Comment (jaeopt) [inline core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfig.java]: I meant to remove @JsonIgnore redundant below at "getDatafile()".
Is it ok to remove @JsonIgnore for datafile completely?

**Pull request merged**